File: raspberry/c/tcp/src/led.c
/*
* Copyright(c) 2016 www.tarchcloud.com. All rights reserved 
* Created by Jason.Chen
*/

#include "../include/led.h"
#include <bcm2835.h>
#define POWER_PIN RPI_GPIO_P1_11

/************************************************
	函数:	
		PowerLenBlink
	参数：
		nCount:	LED闪烁多少次
	说明：
		创建上传数据字符串
*************************************************/
int PowerLenBlink(int nCount)
{	
	int bReturn = 1 ;
	int i = 0 ;
	if(bcm2835_init())
	{
		bcm2835_gpio_fsel(POWER_PIN,BCM2835_GPIO_FSEL_OUTP) ;

		for(i = 0 ;i < nCount ;i++)
		{
			printf("BLINK COUNT:%d\n",i );
			bcm2835_gpio_write(POWER_PIN, HIGH); 
			bcm2835_delay(500);
			bcm2835_gpio_write(POWER_PIN, LOW); 
			bcm2835_delay(500);
		}

		bcm2835_gpio_write(POWER_PIN, LOW); 

		bcm2835_close();

	}else 
	{
		bReturn = 0 ;
	}

	return bReturn ;
}

File: raspberry/python/led/led.py
#copyright@tarchcloud.com reserved, 2016
#history:
#2016.08.26, created by Jason.Chen

import RPi.GPIO as GPIO
import time

POWERLED = 11
SLEEPTIME=1

#init GPIO mode
def InitLed(nLed=POWERLED):
	print('InitLed') ;
	GPIO.setmode(GPIO.BOARD) ;
	GPIO.setup(nLed,GPIO.OUT) ;

def LedBlink(Count,Led=POWERLED,SleepTime=SLEEPTIME):
	print('LedBlind') 
	LenNumber = Led ;
	InitLed(LenNumber)
	i = 0 ;
	while i < Count:
		print 'i:',i
 		GPIO.output(LenNumber,GPIO.HIGH)
 		time.sleep(SleepTime)
 		GPIO.output(LenNumber,GPIO.LOW) 
 		time.sleep(SleepTime)
 		i = i+1

 	print 'led blink done'
 	return 